engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/EmailValidator.java

```diff
@@ -104,17 +105,26 @@ private boolean matchPart(String part, Pattern pattern, int maxLength) {
 	private String toAscii(String unicodeString) throws IllegalArgumentException {
 		String asciiString = "";
 		int start = 0;
-		int end = unicodeString.length() <= 63 ? unicodeString.length() : 63;
+		int index = getBreakingIndex( unicodeString, ASCII_MAX_LABEL_LENGTH );
+		int end = unicodeString.length() <=  index ? unicodeString.length() : index;
 		while ( true ) {
 			// IDN.toASCII only supports a max "label" length of 63 characters. Need to chunk the input in these sizes
 			asciiString += IDN.toASCII( unicodeString.substring( start, end ) );
 			if ( end == unicodeString.length() ) {
 				break;
 			}
 			start = end;
-			end = start + 63 > unicodeString.length() ? unicodeString.length() : start + 63;
+			index = getBreakingIndex( unicodeString, start + ASCII_MAX_LABEL_LENGTH );
+			end = index > unicodeString.length() ? unicodeString.length() : index;
 		}
 
 		return asciiString;
 	}
+
+	private int getBreakingIndex(String unicodeString, int index) {
+		if (unicodeString.length() <= index) {
+			return index;
+		}
+		return unicodeString.charAt( index ) == '.' ? Math.max( 0, index - 1 ) : index;
```

Comment:
So, as I understand it, there is an issue if the string ends with a dot, but not if the string starts with a dot? It would be nice to check this case too.